modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -880,6 +979,22 @@ MechanicalContactConstraint::computeQpOffDiagJacobian(Moose::ConstraintJacobianT
                 return -_test_master[_i][_qp] * penalty * _phi_slave[_j][_qp] * pinfo->_normal(_component) * normal_component_in_coupled_var_dir;
               else
                 return 0;
+            case CF_TANGENTIAL_PENALTY:
+            {
+              if (pinfo->_mech_status == PenetrationInfo::MS_SLIPPING ||
+                  pinfo->_mech_status == PenetrationInfo::MS_SLIPPING_FRICTION)
+              {
+                RealVectorValue jac_vec;
+                for (unsigned int i=0; i<_mesh_dimension; ++i)
+                {
+                  dof_id_type dof_number = _current_node->dof_number(0, _vars(i), 0);
+                  jac_vec(i) = (*_jacobian)(dof_number, _connected_dof_indices[_j]);
+                }
+                return pinfo->_normal(_component)*(pinfo->_normal*jac_vec) * _test_master[_i][_qp];
+              }
+              else
+                return 0;
```

Comment:
I'd prefer this to read `return 0.0;`  to make it obvious that the function is returning a floating point number